src/lib/accessibility/FocusHandling/AtomicItemFocusHandler.tsx

```diff
@@ -0,0 +1,109 @@
+export interface IAtomicItemProps {
+  isFocused: boolean
+
+  isFirstElement: boolean
+  isLastElement: boolean
+
+  onMovePrevious: () => void
+  onMoveNext: () => void
+  onMoveFirst: () => void
+  onMoveLast: () => void
+  onEnter: () => void
+  onSpace: () => void
+  onEsc: () => void
+}
+
+export interface IAtomicItemState {
+  shouldSubContainerBeOpened: boolean
+  isLastOpened: boolean
+}
+
+export type SetStateDelegate<P, S> = <K extends keyof S>(
+  state:
+    | ((prevState: Readonly<S>, props: Readonly<P>) => Pick<S, K> | S | null)
+    | (Pick<S, K> | S | null),
+  callback?: () => void,
+) => void
+
+export class AtomicItemFocusHandler<P extends IAtomicItemProps, S extends IAtomicItemState> {
+  constructor(
+    private getProps: () => P,
```

Comment:
Is there any reason why not just passing props by wrap it in function?

Reply:
yes - otherwise just reference will be passed, so it won't be updated with update of component props